### PR TITLE
ENG-2102: Filter 'token' values within logs of HTTP requests

### PR DESCRIPTION
**Pull request by JuubiSnake**
This PR:
* Filters 'token' params from logs and redacts them, labelling the value of each param as [FILTERED]
* Patches rails CommonLogger in order to filter tokens that are included within a HTTP request, since `filter_parameters` does not apply to this class.

This references https://jira.fundingcircle.com/browse/ENG-2102

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'rails', '~> 4.2.11'
-
+gem 'rack'
```
Comment (fussybeaver) [inline Gemfile]: Since we're monkey patching rack, consider pegging the library at whichever version we are monkey patching.

**Comment by JuubiSnake** [comment]
A quick preview with a dummy token when calling the `event` endpoint via curl:

Screenshot 2019-06-21 at 16 14 29

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@
     mocks.verify_partial_doubles = true
   end
 
-  config.disable_monkey_patching!
+  # config.disable_monkey_patching!
```
Comment (sgerrand) [inline spec/spec_helper.rb]: 🤔 Was there a reason to do this?

File: config/initializers/rack_common_logger.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# rubocop:disable all
+
+require 'rack/utils'
+
+# We are going to monkey patch the commonlogger that rack uses in order to
```
Comment (sgerrand) [inline config/initializers/rack_common_logger.rb]: 🙋 You can swap the logger than Rails/Rack uses at runtime:
> `config.logger` accepts a logger conforming to the interface of `Log4r` or the default Ruby `Logger` class. Defaults to an instance of `ActiveSupport::Logger`.

https://guides.rubyonrails.org/v4.2/configuring.html

You can also [configure which logger is used via Rack middleware](https://guides.rubyonrails.org/v4.2/configuring.html#configuring-middleware) but I wouldn't recommend doing so for logging of this kind.

File: spec/spec_helper.rb (hunk 2 above)
Comment (JuubiSnake) [reply spec/spec_helper.rb]: Testing a few things - accidentally pushed with this in, so I'll revert it back

**Comment by LS80** [comment]
We can close this PR. The logging middleware needs to be removed to stop swallowing the Octokit client errors, which means we will no longer be logging the request at all.

**Pull request closed without merging**